src/Tools/ExternalAccess/VisualDiagnostics/Contracts/IVisualDiagnosticsLanguageService.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.ServiceHub.Framework;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.VisualDiagnostics.Contracts
+{
+    /// <summary>
+    /// Process Information
+    /// </summary>
+    /// <param name="ProcessId">Unique GUID that uniquely identify a process under a debug session</param>
+    /// <param name="LocalProcessId">local process running on a host device, if the process running on a different host (like mobile device), this will be null</param>
+    /// <param name="Path">path to the process, this is guaranteed to not be null for local process, but could be null for mobile devices</param>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0016:Add public types and members to the declared API", Justification = "False Positive, record is not well supported by the analyzer")]
+    public record struct ProcessInfo(Guid ProcessId, uint? LocalProcessId, string? Path);
```

Comment:
Why any classes or interfaces here need to be public? The pattern used in EA.Xaml is that all classes/interfaces are internal and then it adds `<InternalsVisibleTo Include="CodeAnalysis.csproj"/>` to EA.Xaml.csproj